ert3/engine/_run.py

```diff
@@ -203,17 +220,23 @@ def _prepare_storage_records(
     ensemble_size: int,
     experiment_config: ert3.config.ExperimentConfig,
     parameters_config: ert3.config.ParametersConfig,
+    stages_config: ert3.config.StagesConfig,
     workspace_root: pathlib.Path,
     experiment_name: str,
 ) -> None:
+    stage = stages_config.step_from_key(ensemble.forward_model.stage)
+
     for input_record in ensemble.input:
         record_name = input_record.record
         record_source = input_record.source.split(_SOURCE_SEPARATOR, maxsplit=1)
-
+        record_mime = next(
+            input_.mime for input_ in stage.input if input_.record == record_name
+        )
```

Comment:
Not really the concern of this PR, but I think this suggests that lists might not be the best data structure for holding the inputs :slightly_smiling_face:

Reply:
Wholeheartedly agree, but yes, not really addressable within the scope of this PR.